src/windshaft/client.js

```diff
@@ -169,4 +176,59 @@ WindshaftClient.prototype._jsonpCallbackName = function (payload) {
   return '_cdbc_' + util.uniqueCallbackName(payload);
 };
 
+WindshaftClient.prototype._canPerformRequest = function (request) {
+  return this._requestTracker.canRequestBePerformed(request);
+};
+
+WindshaftClient.prototype._trackRequest = function (request, response) {
+  this._requestTracker.track(request, response);
+};
+
+WindshaftClient.prototype.performRequest = function (payload, params, options) {
+  var request = new Request(payload, params, options);
+  if (this._canPerformRequest(request)) {
+    this._performRequest(request);
+  } else {
+    log.error('Maximum number of subsequent equal requests to the Maps API reached (' + MAP_INSTANTIATION_LIMIT + '):', payload, params);
+    options.error && options.error();
+  }
+};
+
+/**
```

Comment:
Should we remove this? Perhaps we can add JSDoc to the public method and document it a bit.